diff --git a/src/main/java/com/github/demo/model/Book.java b/src/main/java/com/github/demo/model/Book.java
index 8b6041e..6ab4a50 100644
--- a/src/main/java/com/github/demo/model/Book.java
+++ b/src/main/java/com/github/demo/model/Book.java
@@ -11,10 +11,13 @@ public class Book {
 
     private String cover;
 
+    private int rating;
+
     public Book() {
         this.title = "";
         this.author = "";
         this.cover = "";
+        this.rating = 0;
     }
 
     public Book(String author, String title) {
@@ -30,6 +33,14 @@ public class Book {
         this.cover = cover;
     }
 
+    public Book(String author, String title, String cover, int rating) {
+        this();
+        this.author = author;
+        this.title = title;
+        this.cover = cover;
+        this.rating = rating;
+    }
+
     public String getTitle() {
         return title;
     }
@@ -57,4 +68,12 @@ public class Book {
     public void setCover(String cover) {
         this.cover = cover;
     }
+
+    public int getRating() {
+        return rating;
+    }
+
+    public void setRating(int rating) {
+        this.rating = rating;
+    }
 }
diff --git a/src/main/java/com/github/demo/service/BookDatabaseImpl.java b/src/main/java/com/github/demo/service/BookDatabaseImpl.java
index 6b3ab54..e5aa65c 100644
--- a/src/main/java/com/github/demo/service/BookDatabaseImpl.java
+++ b/src/main/java/com/github/demo/service/BookDatabaseImpl.java
@@ -61,7 +61,8 @@ public class BookDatabaseImpl implements BookDatabase {
                 Book book = new Book(
                     rs.getString("author"),
                     rs.getString("title"),
-                    rs.getString("image")
+                    rs.getString("image"),
+                    rs.getInt("rating")
                 );
                 books.add(book);
             }
@@ -89,7 +90,8 @@ public class BookDatabaseImpl implements BookDatabase {
                 Book book = new Book(
                     results.getString("author"),
                     results.getString("title"),
-                    results.getString("image")
+                    results.getString("image"),
+                    results.getInt("rating")
                 );
 
                 books.add(book);
diff --git a/src/main/webapp/books.html b/src/main/webapp/books.html
index 1739dd4..0db57ea 100644
--- a/src/main/webapp/books.html
+++ b/src/main/webapp/books.html
@@ -80,6 +80,12 @@
                   <h4 class="media-heading"><a href="#">[[${book.title}]]</a></h4>
                   <h5 class="media-heading"> by <a href="#">[[${book.author}]]</a></h5>
                   <span>Status: </span><span class="text-success"><strong>In Stock</strong></span>
+                  <!-- Rating shown in stars -->
+                  <div>
+                    <span th:each="i : ${#numbers.sequence(1, book.rating)}">
+                      <img src="/images/star.png" style="width: 12px; height: 12px;"/>
+                    </span>
+                  </div>
                 </div>
               </div>
             </td>
